lib/index.js

```diff
@@ -258,7 +302,8 @@ class Unifile {
    * @return {external:Promise<string>} a promise of the content of the file
    */
 	readFile(session, connectorName, path) {
-		return this.callMethod(connectorName, session, 'readFile', path);
+		return this.callMethod(connectorName, session, 'readFile', path)
+		.then((content) => applyPlugin(this.plugins[HOOKS.ON_READ], content));
```

Comment:
here maybe the plugins should have a ON_BEFORE_READ in order to change the paths and possibly the session and connectorName